File: backend/src/test/java/org/globe42/web/security/PasswordDigesterTest.java
package org.globe42.web.security;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.Test;

/**
 * Unit tests for {@link PasswordDigester}
 * @author JB Nizet
 */
public class PasswordDigesterTest {
    @Test
    public void hashedPasswordShouldMatch() {
        String password = "password";
        PasswordDigester digester = new PasswordDigester();
        String hash = digester.hash(password);
        assertThat(digester.match(password, hash)).isTrue();
    }

    @Test
    public void passwordShouldNotMatchWithNullHashedPassword() {
        PasswordDigester digester = new PasswordDigester();
        assertThat(digester.match("foobar", null)).isFalse();
    }

    @Test
    public void badPasswordShouldNotMatch() {
        String password = "password";
        PasswordDigester digester = new PasswordDigester();
        assertThat(digester.match("badPassword", digester.hash(password)));
    }

    @Test
    public void hashesShouldBeDifferentWithSamePassword() {
        String password = "password";
        PasswordDigester digester = new PasswordDigester();
        String hash1 = digester.hash(password);
        String hash2 = digester.hash(password);
        assertThat(hash1).isNotEqualTo(hash2);
    }
}

File: backend/src/test/kotlin/org/globe42/web/security/PasswordDigesterTest.kt
package org.globe42.web.security

import org.assertj.core.api.Assertions.assertThat

import org.junit.jupiter.api.Test

/**
 * Unit tests for [PasswordDigester]
 * @author JB Nizet
 */
class PasswordDigesterTest {
    @Test
    fun hashedPasswordShouldMatch() {
        val password = "password"
        val digester = PasswordDigester()
        val hash = digester.hash(password)
        assertThat(digester.match(password, hash)).isTrue()
    }

    @Test
    fun passwordShouldNotMatchWithNullHashedPassword() {
        val digester = PasswordDigester()
        assertThat(digester.match("foobar", null)).isFalse()
    }

    @Test
    fun badPasswordShouldNotMatch() {
        val password = "password"
        val digester = PasswordDigester()
        assertThat(digester.match("badPassword", digester.hash(password)))
    }

    @Test
    fun hashesShouldBeDifferentWithSamePassword() {
        val password = "password"
        val digester = PasswordDigester()
        val hash1 = digester.hash(password)
        val hash2 = digester.hash(password)
        assertThat(hash1).isNotEqualTo(hash2)
    }
}
